cypress/e2e/spec.cy.js

```diff
@@ -1,6 +1,58 @@
-// example that will be replaced by actual e2e tests
-describe('template spec', function() {
-    it('passes', () => {
-        assert(true)
-    })
-  })
\ No newline at end of file
+it('Chart is generated', function () {
+    cy.visit('/')
+    cy.findAllByText("Line").click()
+    cy.findAllByLabelText("Chart title").type('Cat super powers')
+    cy.findAllByLabelText("X label").type('cats')
+    cy.findAllByLabelText("Y label").type('powers')
+    cy.findAllByLabelText("X").type('1')
+    cy.findAllByLabelText("Y").type('3')
+    cy.findAllByText("Generate chart").click()
+    cy.get('#chart-img').should('exist')
+})
+
+it('Data is saved between tabs', function () {
+    cy.visit('/')
+    cy.findAllByText("Line").click()
+    cy.findAllByLabelText("Chart title").type('Cat super powers')
+    cy.findAllByLabelText("X label").type('cats')
+    cy.findAllByLabelText("Y label").type('powers')
+    cy.findAllByLabelText("X").type('6')
+    cy.findAllByLabelText("Y").type('9')
+    cy.findAllByLabelText("X").first().invoke('val').should('equal', '6')
```

Comment:
I think you need to click "Scatter" after clicking "Line" and then assert that the data is the same here. Should just be one more line, right after you type everything. Just to make sure that it stays the same when you switch!

Reply:
you right, I also forgot to generate the plot bc I think it saves it?? idk, but it doesn't hurt it. changed on incoming commit